vim-patch:8.1.{560,1204,1206}

janlazo opened the pull request:
```
$ ./scripts/vim-patch.sh -l -- src/testdir/test_usercommands.vim
Updating Vim sources: /home/jan/repo/git/neovim/.vim-src
Already up-to-date.
✔ Updated Vim sources.
Vim patches missing from Neovim:
  • v8.1.1241: Ex command info contains confusing information
```

~~Patch 8.1.0341 is not a full port because it requires v8.0.x patches.~~ Patch v8.1.0341 should be a full port to not fail `test_autocmd.vim` tests so it cannot be included in this PR. Patch v8.1.1241 requires patch v8.1.1210 so it will be ported in a separate PR.

janlazo committed:
vim-patch:8.1.1204: output of :command with address completion is not nice

Problem:    Output of :command with address completion is not nice.
Solution:   Shorten the address completion names.
https://github.com/vim/vim/commit/a561a41a70db7a9367f883c1dbb14e69b3364d08

janlazo committed:
vim-patch:8.1.0560: cannot use address type "other" with with user command

Problem:    Cannot use address type "other" with with user command.
Solution:   Add "other" to the list. (Daniel Hahler, closes vim/vim#3655)  Also
            reject "%" for commands with "other".  Add some more tests.
https://github.com/vim/vim/commit/51a7454cd2c34ee459f104e45768eab3817c099e

janlazo committed:
vim-patch:8.1.1206: user command parsing and listing not properly tested

Problem:    User command parsing and listing not properly tested.
Solution:   Add more tests. (Dominique Pelle, closes vim/vim#4296)
https://github.com/vim/vim/commit/d1f90bbcab0ad934d3f48a717a916807b57269ab

mhinz commented on src/nvim/testdir/test_usercommands.vim:
```diff
@@ -360,7 +360,7 @@ func Test_addr_all()
     endif
   endfor
-  call assert_equal(low, g:a1)
+  " call assert_equal(low, g:a1)
```
I looked through the patch and found no obvious typo.

Then I added this to the test:

```diff
diff --git i/src/nvim/testdir/test_usercommands.vim w/src/nvim/testdir/test_usercommands.vim
index 841c72382..ebadf8d8d 100644
--- i/src/nvim/testdir/test_usercommands.vim
+++ w/src/nvim/testdir/test_usercommands.vim
@@ -360,6 +360,9 @@ func Test_addr_all()
       break
     endif
   endfor
+  call assert_equal(filter(range(1, bufnr('$')), 'buflisted(v:val)'), 999)
+  call assert_equal(filter(range(1, bufnr('$')), 'bufloaded(v:val)'), 999)
+  call assert_equal(g:a1, 999)
   " call assert_equal(low, g:a1)
   call assert_equal(bufnr('$'), g:a2)
```

which resulted in:

```
        function RunTheTest[37]..Test_addr_all line 19: Expected [3, 4, 5] but got 999
        function RunTheTest[37]..Test_addr_all line 20: Expected [3] but got 999
        function RunTheTest[37]..Test_addr_all line 21: Expected 2 but got 999
```

So, `low` is correctly set to 3, but `g:a1` (and thus `<line1>`) is 2 for some reason, although the first listed (and loaded) buffer is 3.

janlazo commented on src/nvim/testdir/test_usercommands.vim:
```diff
@@ -312,3 +340,201 @@ func Test_use_execute_in_completion()
+  args one two three
+  %DoSomething
+  call assert_equal(1, g:a1)
+  call assert_equal(3, g:a2)
```
https://github.com/neovim/neovim/pull/11318/files#diff-4aa0a7f67d049bd357dce116b348d1a1R350-R354
Commenting these out makes the test pass.

janlazo commented:
8.1.0341 resolves the test error. Moved 8.1.0560, 8.1.0573 to https://github.com/neovim/neovim/pull/11873

janlazo commented:
Passes on my machine now for some reason.

janlazo committed:
test/old: skip Test_addr_all() for now

It needs patch v8.1.0341 to pass.
Patch v8.1.0341 fails Test_nocatch_wipe_all_buffers().
